File: server/src/main/kotlin/schema/Cognizance.kt
//
//  Cognizance.kt
//  incicord
//
//  Created by d-exclaimation on 4:15 PM.
//

package schema

import java.time.ZoneId
import java.time.ZoneOffset
import java.time.ZonedDateTime

/**
 * ### Cognizance of connection
 *
 * Confirmation on connection and closing
 */
data class Cognizance(
    val validation: Boolean,
    val timestamp: String
) {
    companion object {
        /**
         * Valid Cognizance Computed Data
         */
        val valid: Cognizance
            get() = Cognizance(
                validation = true,
                timestamp = ZonedDateTime.now(ZoneId.ofOffset("", ZoneOffset.UTC)).toString()
            )

        /**
         * Invalid Cognizance Computed Data
         */
        val invalid: Cognizance
            get() = Cognizance(
                validation = false,
                timestamp = "Invalid Cognizance has no timestamp"
            )
    }
}

File: web/src/models/Cognizance.ts
//
//  Cognizance.ts
//  web
//
//  Created by d-exclaimation on 22:53.
//

export type Cognizance = {
  validation: boolean;
  timestamp: string;
};

export function Cognizance(json: string): boolean {
  const c: Cognizance = JSON.parse(json);
  console.log(new Date(c.timestamp));
  return c.validation;
}
